### PR TITLE
Initial Implementation of Information Dashboard

### PR DESCRIPTION
**Pull request by Hemansh31**
Fixes https://github.com/camicroscope/Distro/issues/95
This is an initial implementation of the Information Dashboard.
image
**Few Ideas**
Clicking on each slide in the table could open a popup with all the summary related to the slide.
@birm Please suggest what else to include.

### CODE REVIEW

**Comment by birm** [comment]
Oh, cool; just looks like you have a minor style fix, and you're good :)

**Comment by Hemansh31** [comment]
@birm what style fix? 
and should I also implement the popup feature I mentioned now.

File: core/Store.js [hunk 1]
```diff
@@ -171,6 +171,16 @@ class Store {
       mode: 'cors',
     }).then(this.errorHandler).then((x)=>this.filterBroken(x, 'mark'));
   }
+  fetchMark(slide_id) {
```
Comment (birm) [inline core/Store.js:174]:
```suggestion
  fetchMark(slideId) {
```

File: core/Store.js [hunk 2]
```diff
@@ -171,6 +171,16 @@ class Store {
...
+    const suffix = 'Mark/find';
+    const url = this.base + suffix;
+    const query = {};
+    query['provenance.image.slide'] = slide_id;
```
Comment (birm) [inline core/Store.js:178]:
```suggestion
    query['provenance.image.slide'] = slideId;
```

File: core/Store.js [hunk 3]
```diff
@@ -254,6 +264,16 @@ class Store {
       mode: 'cors',
     }).then(this.errorHandler).then((x)=>this.filterBroken(x, 'heatmap'));
   }
+  fetchHeatMap(slide_id) {
```
Comment (birm) [inline core/Store.js:267]:
```suggestion
  fetchHeatMap(slideId) {
```

File: core/Store.js [hunk 4]
```diff
@@ -254,6 +264,16 @@ class Store {
...
+    const suffix = 'Heatmap/find';
+    const url = this.base + suffix;
+    const query = {};
+    query['provenance.image.slide'] = slide_id;
```
Comment (birm) [inline core/Store.js:271]:
```suggestion
    query['provenance.image.slide'] = slideId;
```

**Comment by Hemansh31** [comment]
I mean Clicking on each slide in the table could open a popup with all the summary related to the slide.

**Comment by birm** [comment]
That sounds reasonable to me!

**Comment by birm** [comment]
It's up to you if you want us to merge this for now, and do that later, or include that as part of the same PR

**Comment by Hemansh31** [comment]
@birm I prefer to include the feature as a new PR, you could merge this PR for now

**Comment by Vedant1202** [comment]
Hi @Hemansh31, 
The solution you've provided is a good quick fix to the feature request and even occured to me at the first instance, however, it would be very heavy on the server when scaled. This is because you are making a request for the heatmap and annotations seperately for each slide.
That means if you have about 10 slides in a db, you will be making 10 request for heatmaps and 10 requests for annotations, i.e., 20 requests in total. Additionally these 20 requests would be hitting the database everytime. 
So, every reload on the page containing 10 slides makes 40 requests in total which is frankly a very bad ratio. This would cause the server to hang up or even crash if scaled.
A good idea is to gather related data at the database layer itself, so that the data needed gets fetched in a single request as I've done here [#332](https://github.com/camicroscope/caMicroscope/pull/332). This also includes the details modal with you and @birm were discussing about. 
A catch in this fix is that the solution requires mongo 4.0 and up to be implemented. I've discussed the same with @birm and hence I've created a corresponding pull request at [#116](https://github.com/camicroscope/Distro/pull/116).

**Pull request merged**